### PR TITLE
Fixes #17648 - Added support for attributes inheritance

### PR DESCRIPTION
**Pull request by ShimShtein**
Changed HostConverter.to_managed to automatically call attribute inheritance mechanism when a new set of attributes is introduced.

### CODE REVIEW

**Comment by lzap** [comment]
Ok UI works, tests are passing locally, but can you add new test to check the update methods in both UI and API?

Adding also a test for UI edit method would be great too. This is very fragile code with regressions :-(

**Comment by lzap** [comment]
CLI works too nice!

**Comment by ShimShtein** [comment]
Tests added

File: test/functional/api/v2/discovered_hosts_controller_test.rb [hunk 1]
```diff
@@ -54,16 +69,42 @@ def test_delete_discovered_host
...
+    # Get the managed host instance from the DB
+    actual = Host.find(host.id)
+
+    assert_equal hostgroup.id, actual.hostgroup_id
+    assert_equal hostgroup.architecture_id, actual.architecture_id
+    assert_equal hostgroup.operatingsystem_id, actual.operatingsystem_id
+    assert_equal hostgroup.medium_id, actual.medium_id
+    assert_equal hostgroup.ptable_id, actual.ptable_id
+    assert_equal hostgroup.domain_id, actual.domain_id
+  end
```
Comment (lzap) [inline test/functional/api/v2/discovered_hosts_controller_test.rb:103]: Mind to extend this with three more fields - environment, puppet and ca? We had similar issues in the past with these too. It's optional, I buy it just like that too :-)

File: test/functional/discovered_hosts_controller_test.rb [hunk 2]
```diff
@@ -133,6 +133,29 @@ def test_edit_form_submit_parameters
...
+
+    # Get the managed host instance from the DB
+    actual = Host.find(host.id)
+
+    assert_equal hostgroup.id, actual.hostgroup_id
+    assert_equal hostgroup.architecture_id, actual.architecture_id
+    assert_equal hostgroup.operatingsystem_id, actual.operatingsystem_id
+    assert_equal hostgroup.medium_id, actual.medium_id
+    assert_equal hostgroup.ptable_id, actual.ptable_id
+    assert_equal hostgroup.domain_id, actual.domain_id
```
Comment (lzap) [inline test/functional/discovered_hosts_controller_test.rb:162]: Ditto.

**Comment by ShimShtein** [comment]
Extra checks added

**Comment by lzap** [comment]
You are hero! Thanks a lot.

**Pull request merged**